src/parquet/encodings/dictionary-encoding.h

```diff
@@ -418,6 +432,23 @@ inline void DictEncoder<FixedLenByteArray>::WriteDict(uint8_t* buffer) {
   }
 }
 
+template<typename T>
+inline void DictEncoder<T>::FreeMemory() {}
+
+template<>
+inline void DictEncoder<ByteArray>::FreeMemory() {
+  for (const auto& value : uniques_) {
+    pool_->Free(const_cast<uint8_t*>(value.ptr));
+  }
+}
+
+template<>
+inline void DictEncoder<FixedLenByteArray>::FreeMemory() {
+  for (const auto& value : uniques_) {
+    pool_->Free(const_cast<uint8_t*>(value.ptr));
+  }
+}
```

Comment:
Not sure I follow. Doesn't this hit the vtable for each Free call? Even if the function doesn't do anything, this does add overhead.